bundles/ch.elexis.core.ui.mediorder/src/ch/elexis/core/ui/mediorder/AbstractBillAndCloseMediorderHandler.java

```diff
@@ -26,35 +28,43 @@ public abstract class AbstractBillAndCloseMediorderHandler {
 	protected IStatus billAndClose(IModelService coreModelService, IContextService contextService,
 			ICoverageService coverageService, IBillingService billingService, List<IStockEntry> stockEntries,
 			boolean removeStockEntry) {
-
+		
 		if (stockEntries.isEmpty()) {
 			return Status.OK_STATUS;
 		}
 
 		IStock stock = stockEntries.get(0).getStock();
 		IPerson person = stock.getOwner();
 		if (!person.isPatient()) {
-			return Status.error("Not a patient stock");
+			return Status.error(Messages.Mediorder_inactive_patient_stock);
 		}
 
 		IPatient patient = person.asIPatient();
 		ICoverage coverage = coverageService.getLatestOpenCoverage(patient)
-				.orElse(coverageService.createDefaultCoverage(patient));
+				.orElseGet(() -> coverageService.createDefaultCoverage(patient));
 
 		IEncounter billingEncounter = new IEncounterBuilder(coreModelService, coverage,
 				contextService.getActiveMandator().get()).build();
 		VersionedResource vr = VersionedResource.load(null);
-		vr.update("Verrechnung Patientenbestellung", contextService.getActiveUser().get().getId());
-		billingEncounter.setVersionedEntry(vr);
-		coreModelService.save(billingEncounter);
+		vr.update(Messages.Mediorder_Billing_Text,
+				contextService.getActiveUser().get().getId());
 
 		for (IStockEntry stockEntry : stockEntries) {
 			IArticle article = stockEntry.getArticle();
 			Result<IBilled> result = billingService.bill(article, billingEncounter, stockEntry.getCurrentStock());
-
+			
 			if (!result.isOK()) {
-				return Status.error(result.getCombinedMessages());
+				continue;
+			}
+
+			BillingLaw law = result.get().getEncounter().getCoverage().getBillingSystem().getLaw();
+			if (law.equals(BillingLaw.ORG) || law.equals(BillingLaw.privat)) {
+				IEncounter newbillingEncounter = result.get().getEncounter();
+				newbillingEncounter.setVersionedEntry(vr);
+				coreModelService.save(newbillingEncounter);
```

Comment:
Den Ablauf hier verstehe ich nicht. Du erzeugst oben bereits einen Encounter, versuchst dann darauf zu verrechnen wenn das geht dann prüfst du da das law und erzeugst falls falsch noch einen IEncounter?! Das müssen wir diskutieren!

Reply:
Es geht darum einen Text in die Konsultation zu schreiben.
Wenn ein neuer privater Fall erstellt wird, ändert sich der IEncounter. Um diesen IEncounter verwenden zu können, muss ich ihn neu erstellen, da ich diesen nirgendwo herbekomme.